site/formulas/decoding.md

```diff
@@ -7,14 +7,14 @@ follows.
 1. Check that `s_bytes` is the canonical encoding of a field element, or else abort.
 2. Decode `s_bytes`, a byte-string, into \\(s\\), a field element.
 3. Check that \\(s\\) is non-negative, or else abort.
-4. \\(u_1 \gets 1 - a s^{2} \\).
-5. \\(u_2 \gets 1 + a s^{2} \\).
+4. \\(u_1 \gets 1 + a s^{2} \\).
+5. \\(u_2 \gets 1 - a s^{2} \\).
 6. \\(v \gets adu_1^2 - u_2^2 \\).
```

Comment:
I think `u_1` and `u_2` are actually switched here. The [code](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L266) is

```
        // v == ad(1-as²)² - (1+as²)²            where d=-121665/121666
        let v = &(&(-&constants::EDWARDS_D) * &u2.square()) - &u1_sqr;
```
(And the comment in the code is also wrong.)